tests/cephfs/lib/fscrypt_utils.py

```diff
@@ -676,7 +731,95 @@ def delete():
         return_val = 1 if len(failed_ops) > 0 else 0
         return return_val
 
+    def validate_fscrypt_metadata(self, client, encrypt_path):
+        test_files_list = self.get_file_list(client, encrypt_path)
+        test_file = random.choice(test_files_list)
+        out, _ = client.exec_command(sudo=True, cmd=f"ls -l {test_file}")
+        line = out.split("\n")
+        if "total" in line[0]:
+            line.pop(0)
+
+        line_list = line[0].split(" ")
+
+        test_status = 0
+        for i in range(0, len(line_list)):
+            if i == 0:
+                found = 0
+                pattern = ["d", "-", "r", "w", "x", "."]
+                for i in range(0, len(line_list[0])):
+                    if line_list[0][i] in pattern:
+                        found += 1
+
+                if found == len(line_list[0]):
+                    log.info("File permissions not encrypted")
+                else:
+                    log.error("File permissions could be encrypted:%s", line_list[0])
+                    test_status += 1
+            elif i in [1, 4, 6]:
+                info = {1: "Number of links", 4: "Size_in_bytes", 6: "Date"}
+                res = re.search(r"(\d+)", line_list[i])
+                if res:
+                    log.info("%s info is not encrypted", info[i])
+                else:
+                    log.error("%s info could be encrypted:%s", info[i], line_list[i])
+                    test_status += 1
+            elif i in [2, 3]:
+                cmd = "awk -F':' '{ print $1}' /etc/passwd"
+                out, _ = client.exec_command(sudo=True, cmd=cmd)
+                user_list = out.split("\n")
+
+                if line_list[i] in user_list:
+                    log.info("User info is not encrypted")
+                else:
+                    log.error("User info could be encrypted:%s", line_list[i])
+                    test_status += 1
+            elif i == 5:
+                pattern = [
+                    "Jan",
+                    "Feb",
+                    "Mar",
+                    "Apr",
+                    "May",
+                    "Jun",
+                    "Jul",
+                    "Aug",
+                    "Sep",
+                    "Oct",
+                    "Nov",
+                    "Dec",
+                ]
+                if line_list[i] in pattern:
+                    log.info("Month info is not encrypted")
+                else:
+                    log.error("Month info could be encrypted:%s", line_list[i])
+                    test_status += 1
+            elif i == 7:
+                res = re.search(r"(\d+):(\d+)", line_list[i])
+                if res:
+                    log.info("Time info is not encrypted")
+                else:
+                    log.error("Time info could be encrypted:%s", line_list[i])
+                    test_status += 1
+
```

Comment:
We could break this into multiple methods performing a single objective. In such a way, it is easier to read hence improving the maintainability.